### PR TITLE
Fix some react warnings

### LINKED ISSUE #5860: Cleanup errors/warnings when running `yarn amo:dev`
There are a lot of errors related to invalid props when starting amo or disco locally. We should fix those errors.

### PR DESCRIPTION
**Pull request by willdurand**
Fix #5860 

---

This PR aims at fixing most of the issues displayed when running `yarn amo:dev`.

The `InfoDialog` warnings are fixed in #5880

### CODE REVIEW

File: src/core/middleware/security.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export function getNoScriptStyles(
     `src/${appName}/noscript.css`,
   );
   try {
-    return fs.readFileSync(cssPath);
+    return fs.readFileSync(cssPath, 'utf8');
```
Comment (willdurand) [inline src/core/middleware/security.js]: Don't recall the error but we should really pass the encoding in `readFileSync()`

**Comment by rebmullin** [comment]
actually I am seeing this still when I run disco:

```bash
Warning: Failed prop type: The prop `handleGlobalEvent` is marked as required in `DiscoPaneBase`, but its value is `undefined`.
[1]     in DiscoPaneBase
[1]     in Translate(DiscoPaneBase)
[1]     in Connect(Translate(DiscoPaneBase))
...
[1] Warning: Failed prop type: The prop `hasAddonManager` is marked as required in `AddonBase`, but its value is `undefined`.
[1]     in AddonBase
[1]     in WithInstallHelpers
...
[1] Warning: Failed prop type: The prop `status` is marked as required in `AddonBase`, but its value is `undefined`.
[1]     in AddonBase
[1]     in WithInstallHelpers
[1]     in Connect(WithInstallHelpers)
...
```

File: src/amo/components/LanguageTools/index.js [hunk 2]
```diff
@@ -233,9 +233,9 @@ export class LanguageToolsBase extends React.Component<Props> {
                     </Tr>
                   );
                 })
-              : Array(50).fill(
-                  // eslint-disable-next-line react/jsx-indent
-                  <Tr>
+              : Array.from(Array(50)).map((_, i) => (
+                  // eslint-disable-next-line react/jsx-indent, react/no-array-index-key
+                  <Tr key={`LoadingText-${i}`}>
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/amo/components/LanguageTools/index.js]: yep

**Comment by willdurand** [comment]
I'll follow up with the disco pane, which is being modified in other PRs.

**Pull request merged**